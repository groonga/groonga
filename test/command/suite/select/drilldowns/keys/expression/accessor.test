table_create Tags TABLE_PAT_KEY ShortText

table_create Memos TABLE_HASH_KEY ShortText
column_create Memos tags COLUMN_VECTOR Tags
column_create Memos year COLUMN_SCALAR Int32

load --table Memos
[
{"_key": "Groonga is fast!", "tags": ["full-text-search"], "year": 2019},
{"_key": "Mroonga is fast!", "tags": ["mysql", "full-text-search"], "year": 2019},
{"_key": "Groonga sticker!", "tags": ["full-text-search", "sticker"], "year": 2020},
{"_key": "Rroonga is fast!", "tags": ["full-text-search", "ruby"], "year": 2020},
{"_key": "Groonga is good!", "tags": ["full-text-search"], "year": 2020}
]

select Memos \
  --filter '_id > 0' \
  --drilldowns[tags].keys 'tags,year >= 2020' \
  --drilldowns[tags].output_columns _key[0],_key[1],_nsubrecs
