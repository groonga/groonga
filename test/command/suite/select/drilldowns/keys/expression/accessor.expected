table_create Tags TABLE_PAT_KEY ShortText
[[0,0.0,0.0],true]
table_create Memos TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create Memos tags COLUMN_VECTOR Tags
[[0,0.0,0.0],true]
column_create Memos year COLUMN_SCALAR Int32
[[0,0.0,0.0],true]
load --table Memos
[
{"_key": "Groonga is fast!", "tags": ["full-text-search"], "year": 2019},
{"_key": "Mroonga is fast!", "tags": ["mysql", "full-text-search"], "year": 2019},
{"_key": "Groonga sticker!", "tags": ["full-text-search", "sticker"], "year": 2020},
{"_key": "Rroonga is fast!", "tags": ["full-text-search", "ruby"], "year": 2020},
{"_key": "Groonga is good!", "tags": ["full-text-search"], "year": 2020}
]
[[0,0.0,0.0],5]
select Memos   --filter '_id > 0'   --drilldowns[tags].keys 'tags,year >= 2020'   --drilldowns[tags].output_columns _key[0],_key[1],_nsubrecs
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        5
      ],
      [
        [
          "_id",
          "UInt32"
        ],
        [
          "_key",
          "ShortText"
        ],
        [
          "tags",
          "Tags"
        ],
        [
          "year",
          "Int32"
        ]
      ],
      [
        1,
        "Groonga is fast!",
        [
          "full-text-search"
        ],
        2019
      ],
      [
        2,
        "Mroonga is fast!",
        [
          "mysql",
          "full-text-search"
        ],
        2019
      ],
      [
        3,
        "Groonga sticker!",
        [
          "full-text-search",
          "sticker"
        ],
        2020
      ],
      [
        4,
        "Rroonga is fast!",
        [
          "full-text-search",
          "ruby"
        ],
        2020
      ],
      [
        5,
        "Groonga is good!",
        [
          "full-text-search"
        ],
        2020
      ]
    ],
    {
      "tags": [
        [
          5
        ],
        [
          [
            "_key[0]",
            null
          ],
          [
            "_key[1]",
            null
          ],
          [
            "_nsubrecs",
            "Int32"
          ]
        ],
        [
          "full-text-search",
          false,
          2
        ],
        [
          "mysql",
          false,
          1
        ],
        [
          "full-text-search",
          true,
          3
        ],
        [
          "sticker",
          true,
          1
        ],
        [
          "ruby",
          true,
          1
        ]
      ]
    }
  ]
]
